### PR TITLE
Fix board select trigger after flashing

### PR DESCRIPTION
**Pull request by haslinghuis**
- [x] Fixes: https://github.com/betaflight/betaflight-configurator/issues/2773

image
- [x] Also thanks to @Zuldan who found another edge case: when detecting an board as unofficial target it showed as successfully detected and verified. Now the code also checks if the board exists as target:
```
2022-01-27 @20:53:17 -- Configurator has detected the board: SPRACINGH7RF but no official Betaflight target was found
2022-01-27 @20:53:35 -- Query board information to preselect right firmware
2022-01-27 @20:53:35 -- Configurator has successfully detected and verified the board: MATEKF411
```

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by klutvott123 — APPROVED** [review]

**Review by Asizon — APPROVED** [review]
LGTM!

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by limonspb — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by limonspb — DISMISSED** [review]

**Comment by Zuldan** [comment]
Detect

Detection of FC's without unified targets is working well :) Thanks @haslinghuis

**Review by Asizon — CHANGES_REQUESTED** [review]
Ok tested again. I think it is not bugfix and should go in a separate PR at this point, in your decision.
 Now Exit dfu button is greyed qhen my board is in dfu mode, cant exit from dfu using button:
image

File: src/js/tabs/firmware_flasher.js
Comment (Asizon) [inline src/js/tabs/firmware_flasher.js:1153]: maybe the issue is here and we shoul tigered this at the end of function like before.

**Review by Asizon — CHANGES_REQUESTED** [review]

File: src/js/tabs/firmware_flasher.js
Comment (chmelevskij) [inline src/js/tabs/firmware_flasher.js:810]: Not sure what's the point of using filter here, this is just iteration so maybe `forEach` or even better `some` if that's available is a better option here.

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:810]: Yeah, filter is obvious not working here. Wil try with some instead

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js]: Thanks could reproduce `Exit DFU Mode` not working - fixed.

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js]: Fixed

**Review by limonspb — APPROVED** [review]

**Review by Asizon — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Pull request merged**
